src/SigninRequest.ts

```diff
@@ -109,7 +111,11 @@ export class SigninRequest {
             parsedUrl.searchParams.append("nonce", nonce);
         }
 
-        parsedUrl.searchParams.append("state", this.state.id);
+        let state = this.state.id;
+        if (url_state) {
+            state = `${state}${URL_STATE_DELIMITER}${url_state}`;
```

Comment:
Can we actually append more than one value to the parameter ? (ex: `state=ABDC;value1;value2;value3`)
If yes then it means, here, we rely on the developer to use the proper delimiter to separate things inside **url_state**.
I don't know if it's acceptable or not.

Reply:
I could do what MSAL.js does and join the extra values if you think that's preferrable. The url state is just a string so, as a developer, I would imagine that the string I sent is what I should get back. https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/31fb0a77a6bf9db77a41330fa09a76db4ff030c0/lib/msal-common/src/utils/ProtocolUtils.ts#L104

Reply:
I was just curious to know, but leaving it like this as a string is totally fine 😉
One can even encode a complete object as a string and pass it as the **url_state**.

Reply:
I think you raise a good point and maybe joining the leftover values is the way to go. MSAL is used by a lot of apps so if they do it, it’s probably not a bad idea.

Yeah any string would work as the state so you could easily stringify an object and pass it. For a moment I thought about just stringifying whatever data they pass using the existing state param. But, it’s possible that it wouldn’t be serializable or would be too big for a url

Reply:
Good catch for the SigninResponse.
This part looks good to me now 😉